### PR TITLE
[KMCompiler] Add Triton nonzero_static operator

### PR DESCRIPTION
**Pull request by chenkc2025**
## Summary

Add a Triton CUDA implementation for `nonzero_static`.

The operator returns a static-shaped `int64` index tensor with shape `[size, input.dim()]`. It preserves the C-order output order of `torch.nonzero`, truncates the result when `nnz > size`, and pads the remaining rows with `fill_value` when `nnz < size`.

Implementation highlights:

- Uses an ordered `count + prefix + write` pipeline to preserve output order.
- Adds a single-block fast path for small inputs.
- Adds a small-counts path to avoid a separate cumsum call for common mid-size cases.
- Uses `flag_gems.cumsum` only in the general multi-block fallback; the CUDA launch path does not call `torch.cumsum`.
- Fills only the padding tail instead of filling the whole output tensor.
- Uses specialized kernels for `ndim <= 4`, with a generic fallback for `ndim > 4`.
- Supports non-contiguous inputs, scalar/empty tensors, high-rank tensors, complex tensors, and the `out` overload.

## Testing

Correctness:

```bash
pytest -q tests/test_nonzero_static.py -m nonzero_static
```

Performance:

```bash
pytest -s benchmark/test_nonzero_static.py
```

## Performance Results

| Shape / Args | fp16 Torch ms | fp16 Gems ms | fp16 Speedup | bf16 Torch ms | bf16 Gems ms | bf16 Speedup |
|---|---:|---:|---:|---:|---:|---:|
| `(1024,), ratio=0.0, size=128` | 0.014208 | 0.005280 | 2.691 | 0.014224 | 0.005248 | 2.710 |
| `(1024,), ratio=0.1, size=128` | 0.014368 | 0.005280 | 2.721 | 0.013920 | 0.005280 | 2.636 |
| `(16384,), ratio=0.001, size=128` | 0.014656 | 0.010624 | 1.380 | 0.014560 | 0.010656 | 1.366 |
| `(16384,), ratio=0.1, size=1024` | 0.014112 | 0.010592 | 1.332 | 0.014464 | 0.010784 | 1.341 |
| `(262144,), ratio=0.001, size=1024` | 0.014560 | 0.018112 | 0.804 | 0.014688 | 0.017824 | 0.824 |
| `(262144,), ratio=0.1, size=4096` | 0.014560 | 0.017760 | 0.820 | 0.014496 | 0.018144 | 0.799 |
| `(1048576,), ratio=0.001, size=1024` | 0.018080 | 0.018160 | 0.996 | 0.018048 | 0.018720 | 0.964 |
| `(1048576,), ratio=0.1, size=4096` | 0.018048 | 0.017888 | 1.009 | 0.017888 | 0.018368 | 0.974 |
| `(32, 1024), ratio=0.1, size=1024` | 0.017984 | 0.017664 | 1.018 | 0.017952 | 0.017920 | 1.002 |
| `(128, 4096), ratio=0.01, size=4096` | 0.018752 | 0.017824 | 1.052 | 0.018688 | 0.017824 | 1.048 |
| `(1024, 4096), ratio=0.001, size=1024` | 0.038528 | 0.051296 | 0.751 | 0.038496 | 0.050368 | 0.764 |
| `(1024, 4096), ratio=0.1, size=4096` | 0.036608 | 0.051776 | 0.707 | 0.036544 | 0.050336 | 0.726 |
| `(16, 64, 64), ratio=0.1, size=1024` | 0.018240 | 0.017600 | 1.036 | 0.018272 | 0.018304 | 0.998 |
| `(32, 128, 128), ratio=0.01, size=4096` | 0.018752 | 0.018080 | 1.037 | 0.018784 | 0.017952 | 1.046 |

### Average Speedup

The averages below are calculated from the per-case `Torch latency / Gems latency` ratios reported above.

| Dtype | Arithmetic mean | Geometric mean |
|---|---:|---:|
| fp16 | **1.240x** | **1.129x** |
| bf16 | **1.229x** | **1.121x** |
| Overall (28 results) | **1.234x** | **1.125x** |

### CODE REVIEW

File: tests/test_nonzero_static.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+@pytest.mark.parametrize("dtype", COMPLEX_DTYPES)
+def test_nonzero_static_complex(dtype):
+    x_cpu = torch.zeros((3, 4), dtype=dtype)
+    x_cpu[0, 1] = 1 + 0j
+    x_cpu[2, 3] = 0 + 2j
+    x_gpu = x_cpu.cuda()
+
+    actual = nonzero_static(x_gpu, size=4, fill_value=9)
+    ref_x = utils.to_reference(x_gpu)
+    expected = expected_nonzero_static(ref_x, size=4, fill_value=9)
```
Comment (Caeruleann) [inline tests/test_nonzero_static.py]: Why not use `torch.nonzero_static` as ref?

File: src/flag_gems/__init__.py [hunk 2]
```diff
@@ -536,6 +492,11 @@ def torch_ge(v):
     ("nll_loss_nd_forward", nll_loss_nd_forward),
     ("nonzero", nonzero),
     ("nonzero_numpy", nonzero_numpy),
+    (
+        "nonzero_static",
+        nonzero_static,
+        lambda: hasattr(torch.ops.aten, "nonzero_static"),
```
Comment (Caeruleann) [inline src/flag_gems/__init__.py]: Could you replace this hasattr check with a PyTorch version check instead, as is done elsewhere in the repository?

File: src/flag_gems/__init__.py [hunk 3]
```diff
@@ -536,6 +492,11 @@ def torch_ge(v):
     ("nll_loss_nd_forward", nll_loss_nd_forward),
     ("nonzero", nonzero),
     ("nonzero_numpy", nonzero_numpy),
+    (
+        "nonzero_static",
```
Comment (Caeruleann) [inline src/flag_gems/__init__.py:621]: Please also add the `nonzero_staic.out` overload registration and implement `nonzero_static_out` in FlagGems.

PyTorch exposes this overload in native_functions.yaml:
https://github.com/pytorch/pytorch/blob/88f350b95d56adf919da8ea73e014c6d2320e88f/aten/src/ATen/native/native_functions.yaml#L9329-L9342

**Comment by Caeruleann** [comment]
Please rebase your branch to the latest master.

**Comment by chenkc2025** [comment]
> 请将你的分支重新定位到最新的主节点。

done

File: conf/operators.yaml [hunk 4]
```diff
@@ -6255,6 +6255,18 @@ ops:
       - Tensor
     stages:
       - alpha: '5.3'
+  - id: nonzero_static
```
Comment (Caeruleann) [inline conf/operators.yaml:6258]: add `nozero_static_out` id

File: tests/test_nonzero_static.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+    expected = torch.nonzero_static(ref_x, size=16, fill_value=-1)
+    expected = utils.to_reference(expected)
+
+    with flag_gems.use_gems(include=["nonzero_static"]):
+        actual = torch.nonzero_static(x_gpu, size=16, fill_value=-1)
+
+    utils.gems_assert_equal(actual, expected)
+
+
+@pytest.mark.nonzero_static
```
Comment (Caeruleann) [inline tests/test_nonzero_static.py]:
```suggestion
@pytest.mark.nonzero_static_out
```

File: tests/test_nonzero_static.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+    actual_out = torch.empty((1, 1), device=x_gpu.device, dtype=torch.int64)
+    actual = nonzero_static_out(x_gpu, size=16, fill_value=7, out=actual_out)
+
+    assert actual is actual_out
+    assert actual.dtype == torch.int64
+    assert tuple(actual.shape) == tuple(expected.shape)
+    utils.gems_assert_equal(actual, expected)
+
+
+@pytest.mark.nonzero_static
```
Comment (Caeruleann) [inline tests/test_nonzero_static.py]:
```suggestion
@pytest.mark.nonzero_static_out
```

File: src/flag_gems/__init__.py (hunk 3 above)
Comment (chenkc2025) [reply src/flag_gems/__init__.py:621]: 已在最新提交 08e973009 完成该项：实现 nonzero_static_out，补充 ops/__init__.py 导出和 _FULL_CONFIG 中的 nonzero_static.out 注册，并增加对应的 operators.yaml 元数据和测试覆盖。

**Review by Caeruleann — APPROVED** [review]
